ui/src/track-layout/track-layout-react-utils.tsx

```diff
@@ -117,18 +118,38 @@ export function useLocationTrack(
 }
 
 export function useLocationTracks(
-    ids: LocationTrackId[],
+    ids: LocationTrackId[] | undefined,
     layoutContext: LayoutContext,
     changeTime?: TimeStamp,
 ): LayoutLocationTrack[] {
     return (
         useLoader(
-            () => (ids ? getLocationTracks(ids, layoutContext) : undefined),
+            () => {
+                if (ids === undefined) {
+                    return getAllLocationTracks(layoutContext, changeTime);
+                } else if (ids.length > 0) {
+                    return getLocationTracks(ids, layoutContext, changeTime);
+                } else {
+                    return undefined;
+                }
+            },
             [JSON.stringify(ids), layoutContext.branch, layoutContext.publicationState, changeTime],
         ) || []
     );
 }
 
+export function useAllLocationTracks(
+    layoutContext: LayoutContext,
+    changeTime?: TimeStamp,
+): LayoutLocationTrack[] {
+    return (
+        useLoader(
+            () => getAllLocationTracks(layoutContext, changeTime),
+            [layoutContext.branch, layoutContext.publicationState, changeTime],
+        ) || []
+    );
+}
+
```

Comment:
Näitä on meillä koodissa jo paljonkin, niin en nyt mitenkään Copilottia tästä sakottaisi, mutta joskus nämä `useLoader(...) || []`-pykäelmät on liikarenderöintejä selvittäessä tosi ärsyttäviä :sweat_smile: Kun tuo niin staattiselta näyttävä []-literaali tarkoittaa että "allokoi uusi taulukko", niin tästä tulee sitten joka kutsukerralla uusi olio ulos siihen asti, että tuo tieto saadaan oikeasti ladattua.